cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AuthenticationUtils.java

```diff
@@ -85,28 +93,65 @@ public static void validateClientAuthentication(KafkaClientAuthentication authen
      * @param isOpenShift   Indicates whether we run on OpenShift or not
      */
     public static void configureClientAuthenticationVolumes(KafkaClientAuthentication authentication, List<Volume> volumeList, String oauthVolumeNamePrefix, boolean isOpenShift)   {
+        configureClientAuthenticationVolumes(authentication, volumeList, oauthVolumeNamePrefix, isOpenShift, false);
```

Comment:
Could you please clarfy why where there some many changes needed in this file? It looks a bit like refactoring mostly unrelated to the MirrorMaker2. This is something what is very hard to test given the different combinations etc. So unless there was an actual need, it should go into a separate Pr to make testing and reviews easier.

Reply:
The MM2 CR can have multiple clusters defined in it, each with their own TLS and auth configuration. When the auth/TLS uses secrets, the values of those secrets needs to be made available to the MM2 connectors in the Connect cluster we create. The Kafka file config provider is used so that secret values are not passed in the connector configuration JSON sent from the assembly pod to the Connect cluster - instead the `kafka_connect_config_generator.sh` script creates a `/tmp/strimzi-mirrormaker2-connector.properties` properties file in the Connect cluster pods that contains the values of the required secrets, and the connector configuration JSON references them.
To generate the properties file, all the secrets that the connector needs must be mounted, including the OAuth secrets that have only been passed via env variable `valueFrom`s before this commit. 

The changes in `AuthenticationUtils.configureClientAuthenticationVolumes()` and  `AuthenticationUtils. configureClientAuthenticationVolumeMounts()` add a param to control the creation of volumes and volume mounts for the OAuth secrets and use the `createNewSecretVolume()` and `createNewVolumeMount()` methods to ensure we don't create a volume or volume mount if a secret has already been mounted by a previous cluster in the list.

The `getClientAuthenticationProperties()` method has been added for use in the MM2 operator to set the non-secret auth properties in the connector configuration JSON, and rather than handling all the auth properties twice, the method is reused by the `configureClientAuthenticationEnvVars()` method for creation of the env vars needed by the other operators.

Hope that all makes sense!

Reply:
> The changes in AuthenticationUtils.configureClientAuthenticationVolumes() and AuthenticationUtils. configureClientAuthenticationVolumeMounts() add a param to control the creation of volumes and volume mounts for the OAuth secrets and use the createNewSecretVolume() and createNewVolumeMount() methods to ensure we don't create a volume or volume mount if a secret has already been mounted by a previous cluster in the list.

This is actually something what I spent some effort in the ast to avoid. We had this in the past, but it creates a lot of chaos when you do not know what is mounted where, which feature added or didn't added which volume. So in the last PRs I intentionally moved away from this to instead just add the volume and mount it multiple times if needed but to keep it clear what is needed where.

Reply:
OK - I have updated the code to add volumes and mount them multiple times into separate `/opt/kafka/mm2-*` directories for use by the MM2 connectors and the connect secrets will continue to be mounted into the `/opt/kafka/connect-*` directories.